database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java

```diff
@@ -88,6 +108,17 @@ public void startListening() {
     @Override
     public void cleanup() {
         mSnapshots.removeChangeEventListener(this);
+        notifyDataSetChanged();
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
+    void cleanup(LifecycleOwner source, Lifecycle.Event event) {
+        if (event == Lifecycle.Event.ON_STOP) {
+            cleanup();
+        } else if (event == Lifecycle.Event.ON_DESTROY) {
```

Comment:
Hmmm, I'm not sure what you mean. AFAIK, we'll get an `onDestroy` if one occurred—always.